src/historywork/RunCommandWork.h

```diff
@@ -33,6 +33,17 @@ class RunCommandWork : public BasicWork
                    size_t maxRetries = BasicWork::RETRY_A_FEW);
     ~RunCommandWork() = default;
 
+    bool
+    isDone() const
```

Comment:
Also, I just realized that RunCommandWork is not exactly doing the right thing. When aborted, right now it is still possible for the callback to update Work's member variables (which is what you added in WorkTests), but we shouldn't do this. When work is aborted, its state is reset, so a callback making changes to the state is wrong. So I think instead in WorkTests we could just crank for a bit, then check if the number of remaining processes is as expected. 
(regarding the callback firing post reset, I can fix that separately as it's outside of the scope of this PR)

Reply:
I guess I didn't want to just spin on checking that the process eventually exits because it's a process that just sleeps -- it will exit of its own accord regardless. We really want to check that it got _terminated_.

I also meant to ask about this, I guess: the existing code considers the work aborted once it sends a kill, without confirming process exit. We're preserving (and even enhancing) that behaviour a bit in this PR, with the event-handlers being cancelled in advance of the process exit happening. I wanted to check with you that that seems like the right thing in your mind.

It means core might _think_ a process has exited (or at least any part that's just judging from the cancelled event) before the process has actually exited; but on the other hand it means that same listener is not likely to get stuck waiting for an unkillable process, and can make progress on its own knowing that the process manager has at least tried its best to kill the process and it's now up to the OS to deal with. Not sure what's best.

Reply:
Yeah, I think back then I thought that as long as Work tried its best to kill the process, it can transition into the "aborted" state. Thinking about it now, not sure what else we can do here.. I mean, at some point we'll have to just let the OS figure it out. So I'm fine preserving this behavior, mostly because I don't have a better alternative to offer.

re: tests. Yeah, it's annoying :/ I understand your intent, it's just that we're breaking the Work contract here by allowing mDone to be modified after the abort is done.

Reply:
pushed variant that does not alter public API contracts.